File: WebAPI/WebAPI.Application/DTO/MenuDTO/Menu/MenuDataDto.cs
using WebAPI.Application.DTO.MenuDTO.MenuCategory;
using WebAPI.Application.DTO.MenuDTO.MenuDetails;
using WebAPI.Application.DTO.MenuDTO.MenuItem;

namespace WebAPI.Application.DTO.MenuDTO.Menu;

public class MenuDataDto
{
    public MenuDetailsDto MenuDetails { get; set; }
    public List<MenuItemDto> FoodMenuItems { get; set; }
    public List<MenuCategoryDto> FoodMenuCategories { get; set; }
    public List<MenuItemDto> DrinksMenuItems { get; set; }
    public List<MenuCategoryDto> DrinksMenuCategories { get; set; }
    
}

File: WebAppSSR/src/app/core/http/dto/menu-dto/menu/menu-data.dto.ts
import {MenuDetailsDto} from "../menu-details/menu-details.dto";
import {MenuItemDto} from "../menu-item/menu-item.dto";
import {MenuCategoryDto} from "../menu-category/menu-category.dto";

export interface MenuDataDto {
  menuDetails: MenuDetailsDto;
  foodMenuItems: MenuItemDto[];
  foodMenuCategories: MenuCategoryDto[];
  drinksMenuItems: MenuItemDto[];
  drinksMenuCategories: MenuCategoryDto[];
}
